src/occa/internal/modes/dpcpp/utils.cpp

```diff
@@ -0,0 +1,160 @@
+#include <stdio.h>
+
+#include <occa/internal/modes/dpcpp/utils.hpp>
+#include <occa/internal/modes/dpcpp/device.hpp>
+#include <occa/internal/modes/dpcpp/memory.hpp>
+#include <occa/internal/modes/dpcpp/kernel.hpp>
+#include <occa/internal/modes/dpcpp/stream.hpp>
+#include <occa/internal/modes/dpcpp/streamTag.hpp>
+#include <occa/internal/io.hpp>
+#include <occa/internal/utils/sys.hpp>
+#include <occa/internal/utils/env.hpp>
+#include <occa/core/base.hpp>
+
+namespace occa
+{
+  namespace dpcpp
+  {
+    /* Returns true if any DPC++ device is enabled on the machine */
+    bool isEnabled()
+    {
+      auto device_list = ::sycl::device::get_devices();
+      // The SYCL standard states that, when no arguments are passed to
+      // device::get_devices (default argument is info::device_type::all),
+      // at least one device is returned: i.e. the SYCL host device. 
+      return (device_list.size() > 1);
```

Comment:
Should we check `> 0` instead?
```suggestion
      return (device_list.size() > 0);
```

Reply:
At least one device will always be returned (which is the host device); however, we have excluded this previously in the device registration. The host device can be useful for debugging, so lets change this to zero and I will update the device registration function to include the host device.